app/src/main/java/com/kms/cura/view/PatientSignUpActivity.java

```diff
@@ -0,0 +1,104 @@
+package com.kms.cura.view;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import com.kms.cura.R;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PatientSignUpActivity extends AppCompatActivity implements TextWatcher {
+    private EditText edtFirstName, edtEmail, edtPassword, edtPasswordReenter;
+    private Button btnRegister;
+    private int count = 0;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_patient_sign_up);
+        initEditText();
+        btnRegister = (Button) findViewById(R.id.btnRegister);
+        btnRegister.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //Patient Register Function
+            }
+        });
+
+    }
+
+
+    public void initEditText() {
+        edtFirstName = (EditText) findViewById(R.id.edtFirstName);
+        edtEmail = (EditText) findViewById(R.id.edtEmail);
+        edtPassword = (EditText) findViewById(R.id.edtPassword);
+        edtPasswordReenter = (EditText) findViewById(R.id.edtPasswordReenter);
+
+        edtFirstName.addTextChangedListener(this);
+        edtEmail.addTextChangedListener(this);
+        edtPassword.addTextChangedListener(this);
+        edtPasswordReenter.addTextChangedListener(this);
+
+    }
+
+    public boolean isEmail(String email) {
+        String regex = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,6}$";
+        Pattern pattern = Pattern.compile(regex);
+        Matcher m = pattern.matcher(email);
+        if (m.find()) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean isPassword(String password) {
+        if (password.length() > 5 && password.length() < 17) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+        //Ignore
+    }
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+        //Ignore
+    }
+
+    @Override
+    public void afterTextChanged(Editable s) {
+        boolean checkLegal = true;
+        if ("".equals(getEditTextText(edtFirstName))) {
```

Comment:
This afterTextChanged() method is too cumbersome. Should extract each of the `if` statement to a separated method ( for example boolean validateFirstName()) that handle checking and set error to the EditText and return a boolean to calculate to overall status.